starknet_py/cairo/serialization/data_serializers/cairo_data_serializer.py

```diff
@@ -44,11 +46,11 @@ def serialize(self, data: SerializationType) -> CairoData:
         :param data: data to serialize.
         :return: calldata.
         """
-        return list(self._serialize(data, TransformationContext()))
+        return list(self.serialize_with_context(data, SerializationContext()))
 
     @abstractmethod
-    def _deserialize(
-        self, reader: CalldataReader, context: TransformationContext
+    def deserialize_with_context(
```

Comment:
Linter started complaining about using `_deserialize` in child classes, so I renamed it, but I would rather keep it protected.